### PR TITLE
refactor how we allow collateral such that there is a delay period

### PR DESCRIPTION
**Pull request by wilsoncusack**
had to change some stuff to keep contract size down: thing most immutable do not need to be public

### CODE REVIEW

File: src/interfaces/IPaprController.sol [hunk 1]
```diff
@@ -224,10 +224,10 @@ interface IPaprController {
...
     function setLiquidationsLocked(bool locked) external;
 
-    /// @notice sets whether a collateral is allowed to be used to mint debt
-    /// @dev owner function
-    /// @param collateralConfigs configuration settings indicating whether a collateral is allowed or not
-    function setAllowedCollateral(IPaprController.CollateralAllowedConfig[] calldata collateralConfigs) external;
+    // /// @notice sets whether a collateral is allowed to be used to mint debt
+    // /// @dev owner function
+    // /// @param collateralConfigs configuration settings indicating whether a collateral is allowed or not
+    // function setAllowedCollateral(IPaprController.CollateralAllowedConfig[] calldata collateralConfigs) external;
```
Comment (adamgobes) [inline src/interfaces/IPaprController.sol:230]: is this supposed to be commented?

**Comment by wilsoncusack** [comment]
we're still a bit over deploy size limit but I will work it out once #123 also goes in. Sorry for the large diff here due to trying to get the size down

File: src/interfaces/IPaprController.sol [hunk 2]
```diff
@@ -238,23 +243,21 @@ interface IPaprController {
...
+    /// @param asset address of the ERC721 token
+    function proposedTimestamp(ERC721 asset) external view returns (uint256);
+
     /// @notice if liquidations are currently locked, meaning startLiquidationAuciton will revert
     /// @dev for use in case of emergencies
     /// @return liquidationsLocked whether liquidations are locked
     function liquidationsLocked() external view returns (bool);
 
-    /// @notice boolean indicating whether token0 in pool is the underlying token
-    function token0IsUnderlying() external view returns (bool);
```
Comment (adamgobes) [inline src/interfaces/IPaprController.sol:247]: probably need a subgraph update with this, heads up

**Review by adamgobes — APPROVED** [review]
left a small comment bout subgraph but looks good

File: src/interfaces/IPaprController.sol (hunk 2 above)
Comment (wilsoncusack) [reply src/interfaces/IPaprController.sol:247]: Yup thanks!

**Pull request merged**